_posts/2020-04-18-schedule-your-functions.md

```diff
@@ -0,0 +1,130 @@
+date: 2020-04-18
+image: /images/2020-04-18-schedule-your-functions/accuracy-alarm-clock-analogue.jpg
+categories:
+  - cronjobs
+  - kubernetes
+  - serverless
+  - linux
+  - tutorial
+  - example
+author_staff_member: martin
+dark_background: true
+
+---
+
+In this blog post I will show you how you can run your functions on schedule. We will first set our environment and then we will run the cron-connector with example function which will be executed on a schedule.
+
+One of the things we need very often is our functions to be executed on a schedule. This is where the cron-connector shines. We will showcase how you can use it in this post.
+
+## Motivation
+
+OpenFaaS provides very convenient and flexible way to quickly write and expose your code to be used by external systems. Often times those systems don't have a hook to which you can schedule function execution. Also often times we need to run those functions without external systems, we just want our code to be ran on an interval.
+
+For those purposes [connector-sdk](https://github.com/openfaas-incubator/connector-sdk) was created. It provides the means to quickly adapt different systems to invoke functions. Some of them are [Kafka](https://github.com/openfaas-incubator/kafka-connector), [NATS](https://github.com/openfaas-incubator/nats-connector), [AWS SNS](https://github.com/affix/OpenFaaS-SNS) and [many more](https://docs.openfaas.com/reference/triggers/).
+
+Now we need our functions to to be invoked on an interval. The [cron-connector](https://github.com/openfaas-incubator/cron-connector) exists for those purposes much like Kubernetes provides specific object called `CronJob` which executes code on a schedule or how in Linux we have cron configuration which run scripts on a schedule.
+
+Now lets roll up our sleeves and see how we can execute our OpenFaaS functions on time.
+
+## Prerequisites
+
+Before we start we need a couple of binaries to set up our environment:
+
+* [docker](https://docs.docker.com/get-docker/) - the container runtime used in this post
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start/) - lightweight local Kubernetes cluster which uses containers as nodes
+* [arkade](https://github.com/alexellis/arkade) - one line installation of applications with which we will install OpenFaaS
+* [faas-cli](https://docs.openfaas.com/cli/install/) - the CLI which communicates with the OpenFaaS gateway
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) - the CLI which communicates with your Kubernetes cluster
+
+## Prepare environment
+
+Before showing you how to run our functions on a schedule, first we need to set up the environment with the tools mentioned above.
+
+### Create cluster
+
+First thing we need is running Kubernetes cluster:
+
+```bash
+kind create cluster
+```
+
+Wait for the installation to finish and verify the namespaces are `Active`:
```

Comment:
This part is a little ambiguous to me, what do you mean by Active? what should a user expect? Is it just a zero exit code?  It might be good to tell them what they should expect, in this case I think it is this list
```
NAME                 STATUS   AGE
default              Active   36h
kube-node-lease      Active   36h
kube-public          Active   36h
kube-system          Active   36h
local-path-storage   Active   36h
```

Unless there is an alternative command where the output is a more definitive "Everything is correct"?

Reply:
Yep that is what I meant, though I think checking whether your kubernetes cluster is up and running might be related with the pods in the kube-system more rather than namespaces. I will change that with 1/1 pods in the `kube-system` namespace :+1:

Reply:
Changed to `kubectl get pods kube-system` for the control plane :heavy_check_mark: